gcc/rust/rust-gcc.cc

```diff
@@ -176,6 +175,46 @@ class Gcc_backend : public Backend
 
   Btype *char_type () { return this->make_type (char_type_node); }
 
+  bool const_size_cast (Bexpression *expr, size_t *result)
+  {
+    rust_assert (TREE_CONSTANT (expr->get_tree ()));
+
+    unsigned char buf[sizeof (size_t) + 1];
+    memset (buf, 0, sizeof (buf));
+
+    int ret = native_encode_expr (expr->get_tree (), buf, sizeof (buf), 0);
+    if (ret <= 0)
+      return false;
+
+    size_t *tmp = (size_t *) buf;
+    *result = *tmp;
+    return true;
+  }
+
+  std::string const_size_val_to_string (Bexpression *expr)
+  {
+    rust_assert (TREE_CONSTANT (expr->get_tree ()));
```

Comment:
This is really janky @dkm @tschwinge, the capacity constant in arrays are all constant items so fold-const.c correctly folds it all down to a constant value. But i still need to be able to print the capacity for type checking purposes but this just feels very error-prone to me here.

Also the same with the above we want we need to be able to access the value in the case of Copied arrays where we need to create an array constructor of the capacity and pass the same element.